File: javascript/celebrity_recognition_aksk_demo.js
/**
 * 名人识别检测服务的aksk 方式请求的使用示例
 */
var recognition = require("./ais_sdk/celebrity_recognition");
var utils = require("./ais_sdk/utils");

// 初始化服务的区域信息，目前支持华北-北京一(cn-north-1)、亚太-香港(ap-southeast-1)等区域信息
utils.initRegion("cn-north-1");

var app_key = "*************";
var app_secret = "************";

var filepath = "./data/celebrity-recognition.jpg";
var data = utils.changeFileToBase64(filepath);

// obs链接需要和region区域一致，不同的region的obs资源不共享
demo_data_url = 'https://ais-sample-data.obs.cn-north-1.myhuaweicloud.com/celebrity-recognition.jpg';

recognition.celebrity_recognition_aksk(app_key, app_secret, data, "", 0.48, function (result) {
    console.log(result);
});

recognition.celebrity_recognition_aksk(app_key, app_secret, "", demo_data_url, 0.48, function (result) {
    console.log(result);
});

File: python/celebrity_recognition_aksk_demo.py
# -*- coding:utf-8 -*-
from ais_sdk.utils import encode_to_base64
from ais_sdk.celebrity_recognition import celebrity_recognition_aksk
from ais_sdk.utils import init_global_env

if __name__ == '__main__':
    # Services currently support North China-Beijing 1 (cn-north-1), Asia Pacific-Hong Kong (ap-southeast-1)
    init_global_env('cn-north-1')

    #
    # access image tagging ， post data by ak,sk
    #
    app_key = '*************'
    app_secret = '************'

    # The OBS link should match the region, and the OBS resources of different regions are not shared
    demo_data_url = 'https://ais-sample-data.obs.cn-north-1.myhuaweicloud.com/celebrity-recognition.jpg'

    # call interface use the url
    result = celebrity_recognition_aksk(app_key, app_secret, "", demo_data_url, 0.48)
    print(result)

    # call interface use the file
    result = celebrity_recognition_aksk(app_key, app_secret, encode_to_base64('data/celebrity-recognition.jpg'), '', 0.48)
    print(result)
